tests/unit/test_provider.py

```diff
@@ -195,6 +195,64 @@ async def subtest_success():
                 assert mock_sleep.mock.call_count == 0
                 assert mock_delete_hosts.mock.call_count == 0
 
+        async def subtest_reprovision_missing_one_by_one_error():
+            """Test reprovisioning when provisioning fails."""
+            dummy_reqs = [
+                {"name": "host1", "vcpus": 2, "memory": 4096},
+                {"name": "host2", "vcpus": 2, "memory": 4096},
+                {"name": "host3", "vcpus": 2, "memory": 4096},
+            ]
+            missing_dummy_reqs = [
+                {"name": "host3", "vcpus": 2, "memory": 4096},
+            ]
+
+            call_counts = [0]  # Initialize a list to store the call counts
+
+            def provision_side_effect(*args, **kwargs):
+
+                call_counts[0] += 1  # Increment the call count
+                print(f"CALL NO: {call_counts}\n{call_counts[0]}")
```

Comment:
@dav-pascual I will remove these prints and repush, are commit messages fine? i mean especially the one with feat: does feat make sense there or should I rather use fix?

Reply:
I have changed that to fix. it is not major change